azext_iot/tests/test_iot_digitaltwin_unit.py

```diff
@@ -364,51 +386,162 @@ def test_iot_digitaltwin_property_update_error(self, fixture_cmd, serviceclient_
             build_mock_response(mocker, request.param, payload=payload_list),
-            build_mock_response(mocker, request.param, payload=payload_show)
+            build_mock_response(mocker, request.param, payload=payload_show),
+            build_mock_response(mocker, request.param, payload={}),
+            build_mock_response(mocker, request.param, payload={})
         ]
         service_client.side_effect = test_side_effect
         return service_client
 
-    def test_iot_digitaltwin_monitor_events(self, fixture_cmd, serviceclient):
-        subject.iot_digitaltwin_monitor_events(fixture_cmd, device_id=device_id,
-                                               interface='environmentalSensor', source_model='public',
-                                               yes=True, properties='all', login=mock_target['cs'])
-        args = serviceclient.call_args
-        url = args[0][0].url
-        method = args[0][0].method
-
-        assert method == 'GET'
-        assert '/models/' in url
+    @pytest.mark.parametrize("req", [
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               interface='environmentalSensor',
+                                               source_model='public',
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               interface='environmentalSensor',
+                                               source_model='private', repo_id=mock_pnptarget['repository_id'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               interface='environmentalSensor',
+                                               source_model='device', repo_login=mock_pnptarget['cs'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               source_model='public',
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
+                                               source_model='private', repo_id=mock_pnptarget['repository_id'],
+                                               yes=True, properties='all',
+                                               login=mock_target['cs'])),
+        (digitaltwin_monitor_events_create_req(device_id=device_id,
```

Comment:
we should add a case when an interface to filter is passed in but no device

Reply:
Added